src/main/java/com/epam/aidial/core/controller/ShareController.java

```
@@ -0,0 +1,153 @@
+                        }
+                    });
+                })
+                .onSuccess(response -> context.respond(HttpStatus.OK, response))
+                .onFailure(this::handleServiceError);
+    }
+
+    public Future<?> createSharedResources() {
+        return context.getRequest()
+                .body()
+                .compose(buffer -> {
+                    ShareResourcesRequest request;
+                    try {
+                        String body = buffer.toString(StandardCharsets.UTF_8);
+                        request = ProxyUtil.convertToObject(body, ShareResourcesRequest.class);
+                    } catch (Exception e) {
+                        log.error("Invalid request body provided", e);
+                        throw new IllegalArgumentException("Can't initiate share request. Incorrect body");
+                    }
+
+                    String bucketLocation = BlobStorageUtil.buildInitiatorBucket(context);
+                    String bucket = encryptionService.encrypt(bucketLocation);
+                    return proxy.getVertx().executeBlocking(() -> shareService.initializeShare(bucket, bucketLocation, request));
+                }).onSuccess(response -> context.respond(HttpStatus.OK, response))
+                .onFailure(this::handleServiceError);
+    }
+
+    public Future<?> discardSharedResources() {
+        return context.getRequest()
+                .body()
+                .compose(buffer -> {
+                    ResourceLinkCollection request = getResourceLinkCollection(buffer, Operation.DISCARD);
+                    String bucketLocation = BlobStorageUtil.buildInitiatorBucket(context);
+                    String bucket = encryptionService.encrypt(bucketLocation);
+                    return proxy.getVertx()
+                            .executeBlocking(() -> {
+                                shareService.discardSharedAccess(bucket, bucketLocation, request);
```

Does it silently do nothing if there is no resource?

---

yes, we can throw exception if no resource provided